VictoriaAlbert/VictoriaAlbert/RecordTableViewController.swift

```diff
@@ -0,0 +1,169 @@
+		}
+		else {
+			sortButton.image = #imageLiteral(resourceName: "filter_empty")
+		}
+	}
+	
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		self.title = searchTerm
+		self.searchField.delegate = self
+		loadRecords(searchString: searchTerm)
+	}
+	
+	internal func loadRecords(searchString: String) {
+		let escapedString = searchTerm.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)
+		APIRequestManager.manager.getData(endPoint: "http://www.vam.ac.uk/api/json/museumobject/search?q=\(escapedString!)") { (data) in
+			if data != nil {
+				
+				if let records = Record.getRecords(from: data!) {
+					print("We've got Records! \(records.count)")
+					self.records = records
+					self.countLocations()
+					DispatchQueue.main.async {
+						self.tableView.reloadData()
+					}
+				}
+			}
+		}
+	}
+	
+	internal func countLocations() {
+		locationDict = [String : Int]()
+		locationArray = [String]()
+		for all in records {
+			locationDict[all.place] = 0
+		}
+		
+		locationArray = Array(locationDict.keys).sorted { $0 < $1 }
+		print("\n\n\nLocation dict: \(locationDict.count)")
+		print("\nLocation Array: \(locationArray)")
+	}
+	
+	// MARK: - Table view data source
+	
+	override func numberOfSections(in tableView: UITableView) -> Int {
+		if sorted {
+			return locationDict.count
+		}
+		else {
+			return 1
+		}
+	}
+	
+	override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		if sorted {
+			let recordsByLocation = records.filter { (record) -> Bool in
+				(record.place) == locationArray[section]
+			}
+			
+			return recordsByLocation.count }
+		else {
+			return records.count
+		}
+	}
+	
+	override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+		guard sorted else { return nil }
+		return "\(locationArray[section])"
+	}
+	
+	override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		let cell = tableView.dequeueReusableCell(withIdentifier: "recordCell", for: indexPath)
+		var record: Record?
+		if sorted {
+			let recordsByLocation = records.filter { (record) -> Bool in
+				(record.place) == locationArray[indexPath.section]
+				}.sorted { $0.object < $1.object }
+			
+			record = recordsByLocation[indexPath.row]
+		}
+		else {
+			record = records.sorted { $0.object < $1.object } [indexPath.row]
+		}
+		
+		if let recordFixed = record {
+			cell.textLabel?.text = "\(recordFixed.object), \(recordFixed.date) - \(recordFixed.place)"
+			cell.detailTextLabel?.text = recordFixed.title
+			
+			APIRequestManager.manager.getData(endPoint: recordFixed.imageSmall ) { (data: Data?) in
+				if  let validData = data,
+					let validImage = UIImage(data: validData) {
+					DispatchQueue.main.async {
+						cell.imageView?.image = validImage
+						cell.setNeedsLayout()
+					}
+				}
+			}
+		}
+		
+		return cell
+	}
+	
+	// MARK: - Navigation
+	
```

Comment:
It's not really a problem but it looks like you're not using the same indenting as we are. If it doesn't cause you too much heartache I'd prefer if your indenting was 4 spaces (not tabs).